### ISSUE TITLE
Initialize from webpack

### ISSUE THREAD

**Issue by sytheveenje** [jackocnr/intl-tel-input#802]
Hi Jack,

I installed this library trough NPM and load it in my project like this:

```
import 'intl-tel-input';


var input = document.querySelector("#phone");
window.intlTelInput(input);
```

But I get an `window.intlTelInput is not a function` error.

I tried to wrap it into a jquery document ready, but without any success. Can you provide a best practice for loading this library?

```
import $ from 'jquery';
import 'intl-tel-input/build/js/intlTelInput.js';

$( document ).ready(function() {
    var input = document.querySelector("#phone");
    window.intlTelInput(input);
});

```

**Comment by jackocnr** [comment]
It should work if you do `import intlTelInput from '...'` and then use that variable directly.

**Comment by sytheveenje** [comment]
I tried that, but it doesn't work.

**Comment by jackocnr** [comment]
What's the error if you do that?

**Comment by vivatum** [comment]
@sytheveenje thanks for your question, it helped me. My example is below (Angular 6)

1. package.json
   ` "intl-tel-input": "^14.0.3"`

2. angular.json
    `"styles": [
         "node_modules/intl-tel-input/build/css/intlTelInput.css"
    ]`

3. html
```
<mat-form-field>
<input matInput id="phone" type="tel" formControlName="phoneNumberCtr" (keyup)="onPhoneNumberChange()">
<script src="./../../../../node_modules/intl-tel-input/build/js/intlTelInput.js"></script>
</mat-form-field>
```

4. component.ts
```
import { AfterViewChecked, Component, OnInit } from '@angular/core';
import intlTelInput from 'intl-tel-input';
import '../../../../node_modules/intl-tel-input/build/js/utils.js';

ngAfterViewChecked() {
    if (!this.initialViewCheck) {
      this.initIntTelInput();
      this.initialViewCheck = true;
    }
}

private initIntTelInput() {
    this.inputTel = document.querySelector('#phone');
    this.iti = intlTelInput(this.inputTel, {
      initialCountry: 'gb',
      preferredCountries: ['gb', 'il'],
      utilsScript: '../../../../node_modules/intl-tel-input/build/js/utils.js',
      separateDialCode: true
    });
    this.countryChangeListener();
  }

  private countryChangeListener() {
    this.inputTel.addEventListener('countrychange', () => {
      this.onPhoneNumberChange();
    });
  }

  onPhoneNumberChange() {
    this.fullNumber = this.iti.getNumber();
  }
```

**Comment by jackocnr** [comment]
Closing due to inactivity.

**Comment by ptran123** [comment]
Try this.  It works for me.

import * as intlTelInput from 'intl-tel-input';

var input = document.querySelector("#phone");
intlTelInput(input);

**Comment by Apokly** [comment]
For anyone using webpack and having troubles, here is my solution to import stylesheet, using scss file :

```scss
$flagsImagePath: '~intl-tel-input/build/img/';
@import '~intl-tel-input/src/css/intlTelInput';
```

**Comment by Michaelwan777** [comment]
> For anyone using webpack and having troubles loading flags, here is my solution to import stylesheet, using scss file :
> 
> ```css-scss
> $flagsImagePath: '~intl-tel-input/build/img/';
> @import '~intl-tel-input/src/css/intlTelInput';
> ```

You are the best! Great thanks for comments above!

**Comment by Michaelwan777** [comment]
> Stumble upon a related problem - I can initialize an input but can't destroy it:
> 
> ```
> TypeError: intl_tel_input__WEBPACK_IMPORTED_MODULE_0__.getInstance is not a function
> ```
> 
> code:
> 
> ```
> import intlTelInput from 'intl-tel-input';
> // ...
> const input = document.querySelector('#message_recipient_contact');
> intlTelInput.getInstance(input).destroy();
> ```
> 
> also tried `import * as intlTelInput from 'intl-tel-input';`
> 
> intl-tel-input version is 16.0.3
> webpack version is 4.4.0

Here is code works for me:
`import { Component, OnInit } from "@angular/core";
import intlTelInput from "intl-tel-input";
import "../../../../../node_modules/intl-tel-input/build/js/utils.js";

@Component({
  selector: "app-index",
  templateUrl: "./index.component.html",
  styleUrls: ["./index.component.scss"]
})
export class IndexComponent implements OnInit {
  public time: Date = new Date();

  constructor() {}

  private initIntTelInput(): void {
    var inputTel = document.querySelector("#phone");
    var iti = intlTelInput(inputTel, {
      initialCountry: "gb",
      preferredCountries: ["gb", "il"],
      utilsScript:
        "../../../../../node_modules/intl-tel-input/build/js/utils.js",
      separateDialCode: true
    });
  }

  public timer(): void {
    const oneSecond = 1000;
    setInterval(() => {
      this.time = new Date();
    }, oneSecond);
  }

  ngOnInit() {
    this.timer();
    this.initIntTelInput();
  }
}
`

`$flagsImagePath: '~intl-tel-input/build/img/';
@import '~intl-tel-input/src/css/intlTelInput';`

`<input type="tel" id="phone">`

In conclusion, the JS only need to be loaded in each Module/Component which requires this plugin. Don't need to refer js and css in shared index.html. Hope it may help.

**Comment by binDongKim** [comment]
@Michaelwan777 
Did you manage to use `utilsScript` option in Angular?

**Comment by phena109** [comment]
> 
> 
> For anyone using webpack and having troubles loading flags, here is my solution to import stylesheet, using scss file :
> 
> ```css-scss
> $flagsImagePath: '~intl-tel-input/build/img/';
> @import '~intl-tel-input/src/css/intlTelInput';
> ```

Just want to add, this solution works for me but only when I use non-scoped style block. So if you still have trouble when using this solution, try this.

**Comment by xnf** [comment]
> @Michaelwan777 Did you manage to use `utilsScript` option in Angular?

I am writing you from the future, but since i got here via Google, then leaving a note if somebody else gets here. 

I managed to get utilsScript working via trial and error and in the end i just copied it to the assets folder. 

here is a short version of my phone number component (using Angular 17.x whichever version has the viewChild signal)

```
import { ChangeDetectionStrategy, Component, effect, ElementRef, viewChild } from '@angular/core';
import intlTelInput from 'intl-tel-input';

@Component({
	standalone: true,
	selector: 'app-phone-number',
	template: `
		<input type="tel" name="phoneNumber" autocomplete="tel" #intlPhoneElement />
	`,
	changeDetection: ChangeDetectionStrategy.OnPush,
})
export class PhoneNumberComponent {
	intlPhoneElement = viewChild<ElementRef<HTMLInputElement>>('intlPhoneElement');
	iti!: intlTelInput.Plugin;

	constructor() {
		effect(() => {
			const intlPhoneElement = this.intlPhoneElement();
			if (intlPhoneElement) {
				const options: intlTelInput.Options = {
					separateDialCode: true,
					formatOnDisplay: true,
					initialCountry: 'auto',
					nationalMode: true,
					utilsScript: 'assets/intl-tel-input/utils.js',
				};
				this.iti = intlTelInput(intlPhoneElement.nativeElement, options);
				this.iti.promise.then((x) => {
					(window as any).intlTelInputGlobals.loadUtils('/assets/intl-tel-input/utils.js');
				});
			}
		});
	}
}
```
